src/runtime_safe_heap.js

```diff
@@ -8,91 +8,39 @@
 #error "should only be inclded in SAFE_HEAP mode"
 #endif
 
-/** @param {number|boolean=} isFloat */
-function getSafeHeapType(bytes, isFloat) {
-  switch (bytes) {
-    case 1: return 'i8';
-    case 2: return 'i16';
-    case 4: return isFloat ? 'float' : 'i32';
-    case 8: return isFloat ? 'double' : 'i64';
-    default: abort(`getSafeHeapType() invalid bytes=${bytes}`);
-  }
-}
-
 #if SAFE_HEAP_LOG
 var SAFE_HEAP_COUNTER = 0;
 #endif
 
-/** @param {number|boolean=} isFloat */
-function SAFE_HEAP_STORE(dest, value, bytes, isFloat) {
+function SAFE_HEAP_INDEX(arr, idx, log) {
```

Comment:
Maybe call `log` something like `action` since its also just used the abort calls=?